src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala

```diff
@@ -140,7 +140,8 @@ class Routes @Inject() (implicit val materializer: ActorMaterializer,
     }
 
   private def saveRoute: (RouteIn) => Future[Option[RouteOut]] = (route: RouteIn) => {
-    routesService.createRoute(route).flatMap {
+    // TODO use the right parameters
+    routesService.createRoute(route, "", "").flatMap {
```

Comment:
If you wrap both cases of a Future in a another Future, you can also just map it instead of FlatMapping. This will _automagically_ wrap your result in the same monad